modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue

```diff
@@ -99,21 +112,51 @@ export default {
     }
   },
   watch: {
+    'doc.data': {
+      deep: true,
+      handler(newData, oldData) {
+        this.$nextTick(() => {
+          // If the previous state was an empty object, it's not "edited"
+          if (Object.keys(oldData).length > 0) {
+            this.docEdited = true;
+          }
+        });
+      }
+    },
     media(newVal) {
-      this.doc.data = klona(newVal);
-      this.generateLipKey();
+      if (
+        this.docEdited && !this.changeConfirmed &&
+        newVal._id !== this.activeMedia._id
+      ) {
+        this.confirmContent = {
+          heading: 'Unsaved Changes',
+          description: 'Do you want to abandon changes to the active image?',
+          affirmativeLabel: 'Yes'
+        };
+
+        return;
+      }
+
+      this.updateActiveDoc(newVal);
     }
   },
   mounted() {
     this.generateLipKey();
+    this.docEdited = false;
   },
   methods: {
+    updateActiveDoc(newMedia) {
+      this.activeMedia = newMedia;
+      this.doc.data = klona(newMedia);
+      this.generateLipKey();
+    },
     save() {
       this.triggerValidation = true;
       apos.bus.$emit('busy', true);
-      const route = `${this.moduleOptions.action}/${this.media._id}`;
+      const route = `${this.moduleOptions.action}/${this.activeMedia._id}`;
```

Comment:
if `this.moduleOptions` defaults to `{}` does this get all upset?

Reply:
The route would break, but it wouldn't throw an error. If the module has no browser options a bunch wouldn't be working.

Reply:
Yeah, this will never be missing for a piece (or else nothing works).